### PR TITLE
[AIRFLOW-6396] Use tempfile.TemporaryDirectory instead of custom one

### PR DESCRIPTION
**Pull request by turbaszek**
Make sure you have checked _all_ steps below.

### Jira

- [ ] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-6396

### Description

- [ ] Here are some details about my PR, including screenshots of any UI changes:
Removing `airflow.utils.file.TemporaryDirectory` because the same functionality is provided by `tempfile.TemporaryDirectory`.

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [ ] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### CODE REVIEW

File: airflow/utils/file.py [hunk 1]
```diff
@@ -17,32 +17,14 @@
...
-import errno
 import os
 import re
-import shutil
 import zipfile
-from contextlib import contextmanager
-from tempfile import mkdtemp
 from typing import Dict, List, Optional, Pattern
 
 from airflow import LoggingMixin, conf
```
Comment (mik-laj) [inline airflow/utils/file.py:27]: Can you maintain backward compatibility to make migrations easier for users? I think I don't need to take drastic steps and immediately delete this function without warning.  I think that changes should always be done in several stages:

- initial version
- initial version with a deprecation warning
- remove features.

We can make changes to Airflow 2.0 that are not backward compatible, but these should be individual exceptions, but not the only way to make changes.

File: UPDATING.md [hunk 2]
```diff
@@ -41,6 +41,10 @@ assists users migrating to a new version.
 
 ## Airflow Master
 
+### Remove airflow.utils.file.TemporaryDirectory
+
+Removing this because the same functionality is provided by `tempfile.TemporaryDirectory`.
```
Comment (mik-laj) [inline UPDATING.md]: Can you add more information here?
In my opinion, a good note should contain the following information:

- [ ] Previous behaviors
- [ ] New behaviors
- [ ] If possible, a simple example of how to migrate
- [ ] If possible, the benefit for the user after migration e.g. "we want to make these changes to unify class names."
- [ ] If possible, the reason for the change e.g. reference for AIP-21

Here is a similar example that includes most if not all of the requirements:
 https://github.com/apache/airflow/blob/master/UPDATING.md#removal-of-redirect_stdout-redirect_stderr

File: airflow/utils/file.py (hunk 1 above)
Comment (mik-laj) [reply airflow/utils/file.py:27]: We can shorten this path if we suspect that users do not use this function, but I think that this does not apply in this case..  This PR introduces changes to 13 files, so this is a commonly used function.

File: UPDATING.md (hunk 2 above)
Comment (turbaszek) [reply UPDATING.md]: Are those bullets points somewhere available?

Comment (mik-laj) [reply UPDATING.md]: No. This is based on my experience. I would expect the following points, so I try to include them.

File: airflow/utils/file.py (hunk 1 above)
Comment (turbaszek) [reply airflow/utils/file.py:27]: I added deprecation warning.

File: UPDATING.md (hunk 2 above)
Comment (turbaszek) [reply UPDATING.md]: Maybe it's worth to add them to UPDATING.md? ;)

Comment (mik-laj) [reply UPDATING.md]: If you are interested in the subject, I invite you to read the document below, which contains some good examples.
https://developers.google.com/style/inclusive-documentation

> Not recommended: Before launch, give everything a final sanity-check.
> Recommended: Before launch, give everything a final check for completeness and clarity.

Comment (mik-laj) [reply UPDATING.md]: I think it's worth doing. I'll do it in a moment.

Comment (mik-laj) [reply UPDATING.md]: https://github.com/apache/airflow/pull/6960

File: UPDATING.md [hunk 3]
```diff
@@ -41,6 +41,10 @@ assists users migrating to a new version.
 
 ## Airflow Master
 
+### Deprecated airflow.utils.file.TemporaryDirectory
+
+Instead user should use `tempfile.TemporaryDirectory` which provides the same functionality.
```
Comment (mik-laj) [inline UPDATING.md]: Can you add an example?

Comment (turbaszek) [reply UPDATING.md]: I think now the description is sufficient.

**Review by mik-laj — APPROVED** [review]

**Review by feluelle — APPROVED** [review]

**Pull request merged**